src/app.ts

```diff
@@ -13,14 +13,49 @@ interface TemperatureSummary {
   average: number
 }
 
+
+const redingMap: { [key: string]: { [key: string]: number[] } } = {}; 
+
 export function processReadings(readings: TemperatureReading[]): void {
-  // add here your code
+  readings.forEach((reding) => {
+
+    const date = reding.time.toString().split('T')[0];
+    if (!redingMap[date]) {
+      redingMap[date] = {};
+    }
+
+    if (!redingMap[date][reding.city]) {
+      redingMap[date][reding.city] = [];
+    }
+
+    redingMap[date][reding.city].push(reding.temperature);
+
+  });
 }
 
 export function getTemperatureSummary(
   date: Date,
   city: string,
+
 ): TemperatureSummary | null {
-  //add here your code
-  return null
+  const dateString = date.toString().split('T')[0];
+  
+  if (!redingMap[dateString] || !redingMap[dateString][city]) {
+    return null;
+  }
+
+  const temperatures = redingMap[dateString][city];
+  const first = temperatures[0];
```

Comment:
Do you know why the `at` function is not valid here? what would returns if were valid?

Reply:
I have not worked with the at() method. But I have researched that its operation depends on the version used. As of version 16.00 it is supported, in the code const temperatures = redingMap[dateString][city]; temperatures is an array of numbers so according to my research it should be supported. If it were valid it should return the first position of the array.

Reply:
Yes it also because of this https://dev.to/msabir/es2022-brings-at-for-array-4o5o
Also, at() will returns the desired value from the array but if doesn't find the element it would returns an undefined
value